src/app/destination.rs

```diff
@@ -0,0 +1,123 @@
+use http;
+use std::fmt;
+use std::net::SocketAddr;
+
+use proxy::{
+    http::{h1, Settings},
+    Source,
+};
+use transport::{DnsNameAndPort, Host, HostAndPort};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Destination {
+    pub name_or_addr: NameOrAddr,
+    pub settings: Settings,
+    _p: (),
```

Comment:
This is a pretty common idiom to my understanding (for example https://github.com/hyperium/http/blob/master/src/request.rs#L181). It ensures that the type can't be constructed outside of this module and ensures that the type is not used in an exhaustive match situation.

Reply:
Ah, yeah, that makes sense, and I had _figured_ it was something like that. I think it might be marginally clearer if it were called "_priv" or "_private", but I don't have a problem with the current naming.